csrc/tensor_view.cpp

```diff
@@ -1198,17 +1198,43 @@ void TensorView::clearReductionIterDomains() {
       getLeafDomain() == getRootDomain(),
       "should not call clearReductionIterDomains on already transformed TensorDomains");
 
+  std::vector<IterDomain*> root = getRootDomain();
+  std::vector<IterDomain*> alloc = getMaybeAllocationDomain();
```

Comment:
IIUC const& should be good enough. The code below doesn't modify these two variables.